Kernel/TermIterators.hpp

```diff
@@ -44,6 +44,21 @@ class VariableIterator
   DECL_ELEMENT_TYPE(TermList);
   VariableIterator() : _stack(8), _used(false) {}
 
+  VariableIterator& operator=(VariableIterator&&) = default;
```

Comment:
Does this do what you want? I don't think it calls the move-constructor.

In general, this looks like somewhere you might employ the copy-and-swap idiom.

Reply:
You'r right. I thought for some reason it woud use the move constructor if we default. Fixed it. (There was no actual bug because the move assignment was never used anywhere)